.github/workflows/build.yml

```diff
@@ -11,26 +11,36 @@ on:
 
 jobs:
   build-sentry-native:
-    name: sentry-native (${{ matrix.os }})
+    name: sentry-native (${{ matrix.os }}${{ matrix.arch && format('-{0}', matrix.arch) || '' }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        # Pin ubuntu to ensure mono is installed
-        # Pin macos to get the version of Xcode that we need: https://github.com/actions/runner-images/issues/10703
-        os: [ubuntu-22.04, windows-latest, macos-15]
+        include:
+          - os: ubuntu-22.04 # Pin ubuntu to ensure mono is installed
+          - os: macos-15 # Pin macos to get the version of Xcode that we need: https://github.com/actions/runner-images/issues/10703
+          - os: windows-latest
+          - os: windows-11-arm
+            arch: arm64
 
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
       - run: git submodule update --init modules/sentry-native
 
+      # zstd is needed for cross OS actions/cache but missing from windows-11-arm
+      - name: Install zstd on Windows ARM64
+        if: runner.os == 'Windows' && runner.arch == 'ARM64'
+        shell: pwsh
+        run: choco install zstandard
+
       - uses: actions/cache@v4
         id: cache
         with:
           path: src/Sentry/Platforms/Native/sentry-native
-          key: sentry-native-${{ runner.os }}-${{ hashFiles('scripts/build-sentry-native.ps1') }}-${{ hashFiles('.git/modules/modules/sentry-native/HEAD') }}
+          # On Windows, it'll be either: Windows or Windows-arm64:
+          key: sentry-native-${{ matrix.arch && format('{0}-{1}', runner.os, matrix.arch) || runner.os }}-${{ hashFiles('scripts/build-sentry-native.ps1') }}-${{ hashFiles('.git/modules/modules/sentry-native/HEAD') }}
```

Comment:
how about using `matrix.os`?

```yaml
key: sentry-native-${{ matrix.os }}-${{ hashFiles(...) }}-${{ hashFiles(...) }}
```

Reply:
Done, makes it also easier to merge with the musl changes:
- https://github.com/getsentry/sentry-dotnet/pull/4188